File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/party/PartyMemberEjectedMessage.scala





















// Generated on 05/03/2013 10:34:47
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait PartyMemberEjectedMessage extends PartyMemberRemoveMessage {
  override val messageId = 6252

  protected trait Message extends PartyMemberRemoveMessage.Message with Serializable {
    override val messageId = PartyMemberEjectedMessage.messageId

    val kickerId: Int

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeInt(kickerId)
    }

  }

  protected def create(partyId: Int, leavingPlayerId: Int, kickerId: Int): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val partyId = buf.readInt()
    val leavingPlayerId = buf.readInt()
    val kickerId = buf.readInt()

    create(partyId, leavingPlayerId, kickerId)
  }

}

object PartyMemberEjectedMessage extends PartyMemberEjectedMessage {
  case class MessageImpl(partyId: Int, leavingPlayerId: Int, kickerId: Int) extends Message
  type Target = MessageImpl

  override def create(partyId: Int, leavingPlayerId: Int, kickerId: Int) = MessageImpl(partyId, leavingPlayerId, kickerId)

  def apply(partyId: Int, leavingPlayerId: Int, kickerId: Int) = create(partyId, leavingPlayerId, kickerId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/PartyMemberEjectedMessage.java


// Generated on 05/08/2013 19:37:49
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class PartyMemberEjectedMessage extends PartyMemberRemoveMessage {
    public static final int MESSAGE_ID = 6252;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int kickerId;
    
    public PartyMemberEjectedMessage() { }
    
    public PartyMemberEjectedMessage(int partyId, int leavingPlayerId, int kickerId) {
        super(partyId, leavingPlayerId);
        this.kickerId = kickerId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeInt(kickerId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        kickerId = buf.readInt();
        if (kickerId < 0)
            throw new RuntimeException("Forbidden value on kickerId = " + kickerId + ", it doesn't respect the following condition : kickerId < 0");
    }
    
}
